drivers/sensor/tdk/icm42x70/icm42x70.c

```diff
@@ -490,6 +397,7 @@ static int icm42670_turn_on_sensor(const struct device *dev)
 	}
 	LOG_DBG("Set accel full scale to: %d G", data->accel_fs);
 
+#if CONFIG_USE_EMD_ICM42670
```

Comment:
This won't work if there is both an icm42670 and an icm42370 in the system. You need to check if *this* device is an icm42670, not if *any* device is an icm42670.

Reply:
You need to store the result of the DT_COMPAT test in the device config struct, then do a runtime check (not a preprocessor check) in the driver function.

Reply:
Ok, thank you for your comment. I kept both checks: a runtime check and a preprocessor check. This way it allows to build a lighter driver for icm42370 alone, and to call the correct APIs using icm42670 and icm42370 in a same system.

Reply:
Looks good now, thanks.